compiler/rustc_infer/src/infer/outlives/obligations.rs

```diff
@@ -99,7 +99,15 @@ impl<'cx, 'tcx> InferCtxt<'cx, 'tcx> {
         cause: &ObligationCause<'tcx>,
     ) {
         let origin = SubregionOrigin::from_obligation_cause(cause, || {
-            infer::RelateParamBound(cause.span, sup_type)
+            infer::RelateParamBound(
+                cause.span,
+                sup_type,
+                match cause.code.peel_derives() {
+                    ObligationCauseCode::BindingObligation(_, span)
+                    | ObligationCauseCode::ImplicitSizedObligation(_, span) => Some(*span),
+                    _ => None,
+                },
+            )
```

Comment:
I suspect that this might be one of the reasons for the seen performance hit.

Reply:
This seems fairly cheap to me, the span is only used in the error case after all. The rest of the variants are larger, so adding the additional span didn't cause a  size increase